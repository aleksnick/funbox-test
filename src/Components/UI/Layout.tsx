import React from "react";
import Grid from "@material-ui/core/Grid";
import IWithStyles from "../../Models/IWithStyles";
import { withStyles } from "@material-ui/core/styles";

const styles = {
  column: {
    paddingLeft: 4
  }
};

export interface LayoutProps extends IWithStyles {
  sidebar: React.ReactNode;
}

/**
 * Badge
 *
 * @export
 * @class Layout
 * @extends {React.Component<LayoutProps>}
 */
export class Layout extends React.Component<LayoutProps> {
  constructor(props: LayoutProps) {
    super(props);
  }

  render() {
    const { sidebar, classes, children } = this.props;
    return (
      <Grid xs={12} container>
        <Grid className={classes["column"]} item>
          {sidebar}
        </Grid>
        <Grid className={classes["column"]} item xs>
          {children}
        </Grid>
      </Grid>
    );
  }
}

export default withStyles(styles)(Layout);
