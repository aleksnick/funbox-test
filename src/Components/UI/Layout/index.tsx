import React from "react";
import Grid from "@material-ui/core/Grid";
import IWithStyles from "../../../Models/IWithStyles";
import { withStyles } from "@material-ui/core/styles";
import { FlexWrap } from "../../../Models/FlexWrap";

const styles = {
  row: {
    width: "100%",
    height: "100%",
    display: "flex",
    overflow: "hidden",
    flexWrap: FlexWrap.NoWrap
  },
  column: {
    paddingLeft: 4
  }
};

export interface LayoutProps extends IWithStyles {
  sidebar: React.ReactNode;
}

/**
 * Layout
 *
 * @export
 * @class Layout
 * @extends {React.Component<LayoutProps>}
 */
export class Layout extends React.Component<LayoutProps> {
  constructor(props: LayoutProps) {
    super(props);
  }

  render() {
    const { sidebar, classes, children } = this.props;
    return (
      <Grid xs={12} className={classes["row"]}>
        <Grid className={classes["column"]} item>
          {sidebar}
        </Grid>
        <Grid className={classes["column"]} item xs>
          {children}
        </Grid>
      </Grid>
    );
  }
}

export default withStyles(styles)(Layout);
