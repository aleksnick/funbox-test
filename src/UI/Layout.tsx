import React from "react";
import Grid from "@material-ui/core/Grid";

export interface LayoutProps {
  sidebar: React.ReactNode;
}

/**
 * Badge
 *
 * @export
 * @class Layout
 * @extends {React.Component<LayoutProps>}
 */
export default class Layout extends React.Component<LayoutProps> {
  constructor(props: LayoutProps) {
    super(props);
  }

  render() {
    const { sidebar, children } = this.props;
    return (
      <Grid xs={12} container spacing={16}>
        <Grid item>{sidebar}</Grid>
        <Grid item xs>
          {children}
        </Grid>
      </Grid>
    );
  }
}
