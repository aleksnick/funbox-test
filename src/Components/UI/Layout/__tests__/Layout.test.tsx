import React from "react";
import Layout from "../";
import { render } from "enzyme";

describe("Layout", () => {
  describe("Snapshots", () => {
    test("Simple Layout snapshot", () => {
      const sidebar = <div>sidebar</div>;
      const wrapper = render(<Layout sidebar={sidebar}>Layout</Layout>);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
