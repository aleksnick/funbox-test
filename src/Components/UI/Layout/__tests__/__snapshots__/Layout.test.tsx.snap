// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Snapshots Simple Layout snapshot 1`] = `
<div
  class="MuiGrid-container-2 MuiGrid-grid-xs-12-42"
>
  <div
    class="MuiGrid-item-3 Layout-column-1"
  >
    <div>
      sidebar
    </div>
  </div>
  <div
    class="MuiGrid-item-3 MuiGrid-grid-xs-true-30 Layout-column-1"
  >
    Layout
  </div>
</div>
`;
